src/test/java/it/gov/pagopa/reward/service/reward/InitiativesEvaluatorServiceImplTest.java

```diff
@@ -0,0 +1,86 @@
+
+    @Test
+    void evaluateInitiativesBudgetAndRules() {
+
+        // Given
+        RuleEngineService ruleEngineService = Mockito.mock(RuleEngineServiceImpl.class);
+
+        TransactionDTO trx = TransactionDTOFaker.mockInstance(1);
+        List<String> initiatives = List.of("INITIATIVE1");
+
+        InitiativeCounters initiativeCounters = InitiativeCounters.builder().initiativeId("INITIATIVE1").build();
+        UserInitiativeCounters userCounters = UserInitiativeCounters.builder()
+                .userId("USER1")
+                .initiatives(Map.of("INITIATIVE1", initiativeCounters))
+                .build();
+
+        Reward reward = new Reward(new BigDecimal("200"));
+        RewardTransactionDTO rTrx = RewardTransactionDTO.builder()
+                        .rewards(Map.of("INITIATIVE1", reward)).build();
+        Mockito.when(ruleEngineService.applyRules(Mockito.same(trx), Mockito.eq(initiatives), Mockito.eq(userCounters))).thenReturn(rTrx);
+
+        InitiativesEvaluatorService initiativesEvaluatorService = new InitiativesEvaluatorServiceImpl(ruleEngineService);
+
+        // When
+        RewardTransactionDTO result = initiativesEvaluatorService.evaluateInitiativesBudgetAndRules(trx, initiatives, userCounters);
+
+        // Then
+        Assertions.assertNotNull(result);
+    }
+
+    @Test
+    void evaluateWithExhaustedInitiative() {
+
+        // Given
+        RuleEngineService ruleEngineService = Mockito.mock(RuleEngineServiceImpl.class);
+
+        TransactionDTO trx = TransactionDTOFaker.mockInstance(1);
+        List<String> initiatives = List.of("INITIATIVE1", "INITIATIVE2");
+
+        UserInitiativeCounters userCounters = UserInitiativeCounters.builder()
+                .userId("USER1")
+                .initiatives(Map.of(
+                                "INITIATIVE1",
+                                InitiativeCounters.builder().initiativeId("INITIATIVE1").exhaustedBudget(true).build(),
+                                "INITIATIVE2",
+                                InitiativeCounters.builder().initiativeId("INITIATIVE2").exhaustedBudget(false).build()
+                        )
+                )
+                .build();
+
+        Reward reward = new Reward(new BigDecimal("200"));
+        RewardTransactionDTO rTrx = RewardTransactionDTO.builder()
+                .rewards(Map.of("INITIATIVE2", reward)).build();
+        Mockito.when(ruleEngineService.applyRules(trx, List.of("INITIATIVE2"), userCounters)).thenReturn(rTrx);
+
+        InitiativesEvaluatorService initiativesEvaluatorService = new InitiativesEvaluatorServiceImpl(ruleEngineService);
+
+        // When
+        RewardTransactionDTO result = initiativesEvaluatorService.evaluateInitiativesBudgetAndRules(trx, initiatives, userCounters);
+
+        // Then
+        Assertions.assertNotNull(result);
+        Assertions.assertEquals(Map.of("INITIATIVE1", List.of("BUDGET_EXHAUSTED")), result.getInitiativeRejectionReasons());
```

Comment:
you have to check if the INITIATIVE2 has rejections or not!
You are not checking if for some bug also INITIATIVE2 has the rejection reasons...

Reply:
Sorry, probably I missread this line :)
you actualy are checking that result.getInitiativeRejectionReasons is map exactly the map that contains just the INITIATIVE1, it's not necessary the line 56 :)